examples/inspection/plot_permutation_importance.py

```diff
@@ -43,7 +43,9 @@
 from sklearn.datasets import fetch_openml
 from sklearn.model_selection import train_test_split
 
-X, y = fetch_openml("titanic", version=1, as_frame=True, return_X_y=True)
+X, y = fetch_openml(
+    "titanic", version=1, as_frame=True, return_X_y=True, parser="pandas"
+)
```

Comment:
It does not seem to be related to this PR, but the following plot seems broken:

https://output.circle-artifacts.com/output/job/8ae7d25d-7564-4b9d-9473-ea18abb57352/artifacts/0/doc/auto_examples/inspection/plot_permutation_importance.html#tree-s-feature-importance-from-mean-decrease-in-impurity-mdi

The title does not match the data and don't understand what was intended here.

The example in the stable branch looks ok:

https://scikit-learn.org/stable/auto_examples/inspection/plot_permutation_importance.html#tree-s-feature-importance-from-mean-decrease-in-impurity-mdi

Reply:
It is missing the transpose. I probably introduce this bug in the last refactoring of this example. Sorry about that.

Reply:
Oh actually not even. It should only be replaced by a bar plot. I copy paster the `get_feature_names_out` but change the plotting. I will make a PR.

Reply:
I opened https://github.com/scikit-learn/scikit-learn/pull/23272